### PR TITLE
hw-mgmt: thermal: Add hysteresis fot dynamic_minimum change

### PR DESCRIPTION
**Pull request by sholeksandr**
Add temperature histeresis for dynamic_minimum change

Signed-off-by: Oleksandr Shamray <oleksandrs@nvidia.com>

### CODE REVIEW

**Comment by sholeksandr** [comment]
RERUN

**Comment by sholeksandr** [comment]
RERUN

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 1]
```diff
@@ -821,87 +823,78 @@ get_fan_direction()
...
+    array=("$@")
+    size=${#array[@]}
+    for ((i=0; i<size; i++)); do
+        trh_vector[i]=${array[i]}
+    done
+}
 
-	# Get fan direction.
-	get_fan_direction
+config_trh_vector()
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:835]: What it "trh", threshold? If yes, why you don't use just "threshold"?

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 2]
```diff
@@ -821,87 +823,78 @@ get_fan_direction()
...
+        tresh_prev=$tresh
+    done
+
+    # themperaure diff betwin current temperature and last dmin_change
+    temperature_diff=$((ambient-temperature_ambient_tresh_cross))
+
+    # check if fan_dynamic_min changed
+    if [ ! "$fan_dynamic_min_curr" -eq "$fan_dynamic_min" ];
+    then
+        # check if temperature chnge more then hysteresis
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:888]: Fix comment - spell check.

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 3]
```diff
@@ -821,87 +823,78 @@ get_fan_direction()
...
+        if [ $temperature_diff -ge $temp_grow_hyst ]; then
+            fan_dynamic_min=$fan_dynamic_min_curr
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:890]: Do we need hysteresis in bot direction? I though we need it only when temperature is going down.

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 4]
```diff
@@ -821,87 +823,78 @@ get_fan_direction()
...
+        tresh=${trh_vector[i]}
+        if [ "$ambient" -lt "$tresh" ]; then
+            fan_dynamic_min_curr=${trh_vector[$((i+1))]}
+            tresh_next=$tresh
+            break
+        fi
+        tresh_prev=$tresh
+    done
+
+    # themperaure diff betwin current temperature and last dmin_change
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:882]: Fix comment - checkspell. "betwin"

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 5]
```diff
@@ -102,6 +102,8 @@ max_amb=120000
 module_counter=0
 gearbox_counter=0
 lc_counter=0
+temp_grow_hyst=1000
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:105]: Do you think 1C is good? I thought about 2C, but not sure.

File: usr/usr/bin/hw-management-thermal-control.sh (hunk 4 above)
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:882]: "themperaure"

**Comment by sholeksandr** [comment]
RERUN

**Comment by sholeksandr** [comment]
RERUN

File: usr/usr/bin/hw-management-thermal-control.sh [hunk 6]
```diff
@@ -821,87 +823,78 @@ get_fan_direction()
...
+        tresh=${treshold_vector[i]}
+        if [ "$ambient" -lt "$tresh" ]; then
+            fan_dynamic_min_curr=${treshold_vector[$((i+1))]}
+            tresh_next=$tresh
+            break
+        fi
+        tresh_prev=$tresh
+    done
+
+    # temperature diff between current temperature and last dmin_change
```
Comment (vadimp-nvidia) [inline usr/usr/bin/hw-management-thermal-control.sh:882]: Start, please, comment with uppercase.

**Comment by sholeksandr** [comment]
RERUN

**Comment by sholeksandr** [comment]
RERUN

**Pull request merged**